ethcore/src/client/test_client.rs

```diff
@@ -622,7 +623,10 @@ impl BlockChainClient for TestBlockChainClient {
 	}
 
 	fn replay_block_transactions(&self, _block: BlockId, _analytics: CallAnalytics) -> Result<Box<Iterator<Item = Executed>>, CallError> {
-		Ok(Box::new(self.execution_result.read().clone().unwrap().into_iter()))
+		match self.execution_result.read().clone() {
```

Comment:
Since it's a test client I think it would be better to require that the execution result is mocked and fail if it's not instead of falling back to empty iterator by default.

Reply:
Still to be addressed.